controller/array_action/array_mediator_svc.py

```diff
@@ -294,6 +297,51 @@ def get_host_by_host_identifiers(self, initiators):
             logger.debug("can not found host by using initiators: {0} ".format(initiators))
             raise controller_errors.HostNotFoundError(initiators)
 
+    def _get_detailed_hosts_list(self):
```

Comment:
1. no need for this general solution, since the py-cli libs don't (and has no plan to) support loading into memory in chunks.
2. the function is still too long (regardless of the general solution), and should still be divided into two.
3. as I wrote, Munch *is* different (avoids consts, avoids the wrongly-public Host class, conforms to the project's current convention, etc...).

Reply:
think the Reader is correct. It is not open for discussion.
Host object is generic. It doesn't have anything svc related and there is a ticket to generalize get_hosts_for_init where it will be used.

Reply:
what do you mean it's not open for discussion? the review is exactly the place for these discussions.
I don't understand; couldn't find any "get_hosts_for_init" - could you point me to the ticket?

Reply:
2. could you also address this function length issue?

Reply:
Your loop doesn't handle list values. See the reader class to understand

Reply:
please address all numbered issues (with numbers).
I think your last comment is for a different existing conversation (simply adding `value = value.split() if ' ' in value else value` after my `value = value.strip()`)

Reply:
yes, the loop should be in a new (second) function.
but there should also be another new (first) function for the first part (from `self.client.svcinfo.lshost()` to `if not detailed_hosts_list_errors`)

Reply:
Moved loop to separate function
but not agree with the second part. This way the original method will just call two other methods so it is meaningless.